packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -236,8 +267,31 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
 
   /// Builder for iOS-style copy/paste text selection toolbar.
   @override
-  Widget buildToolbar(BuildContext context, Rect globalEditableRegion, Offset position, TextSelectionDelegate delegate) {
+  Widget buildToolbar(
+      BuildContext context,
+      Rect globalEditableRegion,
+      TextSelectionPoint leftTextSelectionPoint,
```

Comment:
I understand the motivation for this change. But it's an API breaking change (since our users may have already subclassed TextSelectionControls. In this case we need to follow the process at https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes and circulate the proposal with the community

Reply:
We also can't call it left/right since RTL is user configurable. startTextSelectionPoint and endTextSelection point.

Reply:
I have thought about this problem. It may be more appropriate to pass the initial data to a subclass and then internally calculate a specific location.
However, for a smooth API transition, I kept the midpoint.